tests/pytests/unit/utils/jinja/test_custom_extensions.py

```diff
@@ -0,0 +1,1088 @@
+        "{{ {}|method_call('fromkeys', ['a', 'b', 'c'], 0)|dictsort }}",
+        dict(opts=minion_opts, saltenv="test", salt=local_salt),
+    )
+    assert rendered == "[('a', 0), ('b', 0), ('c', 0)]"
+
+    # missing object method test
+    rendered = render_jinja_tmpl(
+        "{{ 6|method_call('bit_width') }}",
+        dict(opts=minion_opts, saltenv="test", salt=local_salt),
+    )
+    assert rendered == "None"
+
+
+def test_md5(minion_opts, local_salt):
+    """
+    Test the `md5` Jinja filter.
+    """
+    rendered = render_jinja_tmpl(
+        "{{ 'random' | md5 }}",
+        dict(opts=minion_opts, saltenv="test", salt=local_salt),
+    )
+    assert rendered == "7ddf32e17a6ac5ce04a8ecbf782ca509"
+
+
+def test_sha256(minion_opts, local_salt):
+    """
+    Test the `sha256` Jinja filter.
+    """
+    rendered = render_jinja_tmpl(
+        "{{ 'random' | sha256 }}",
+        dict(opts=minion_opts, saltenv="test", salt=local_salt),
+    )
+    assert (
+        rendered == "a441b15fe9a3cf56661190a0b93b9dec7d04127288cc87250967cf3b52894d11"
+    )
+
+
+def test_sha512(minion_opts, local_salt):
+    """
+    Test the `sha512` Jinja filter.
+    """
+    rendered = render_jinja_tmpl(
+        "{{ 'random' | sha512 }}",
+        dict(opts=minion_opts, saltenv="test", salt=local_salt),
+    )
+    assert rendered == str(
+        "811a90e1c8e86c7b4c0eef5b2c0bf0ec1b19c4b1b5a242e6455be93787cb473cb7bc"
+        "9b0fdeb960d00d5c6881c2094dd63c5c900ce9057255e2a4e271fc25fef1"
+    )
+
+
+def test_hmac(minion_opts, local_salt):
+    """
+    Test the `hmac` Jinja filter.
+    """
+    rendered = render_jinja_tmpl(
+        "{{ 'random' | hmac('secret', 'blah') }}",
+        dict(opts=minion_opts, saltenv="test", salt=local_salt),
+    )
+    assert rendered == "False"
+
+    rendered = render_jinja_tmpl(
+        "{{ 'get salted' | "
+        "hmac('shared secret', 'eBWf9bstXg+NiP5AOwppB5HMvZiYMPzEM9W5YMm/AmQ=') }}",
+        dict(opts=minion_opts, saltenv="test", salt=local_salt),
+    )
+    assert rendered == "True"
+
+
+def test_base64_encode(minion_opts, local_salt):
+    """
+    Test the `base64_encode` Jinja filter.
+    """
+    rendered = render_jinja_tmpl(
+        "{{ 'random' | base64_encode }}",
+        dict(opts=minion_opts, saltenv="test", salt=local_salt),
+    )
+    assert rendered == "cmFuZG9t"
+
+
+def test_base64_decode(minion_opts, local_salt):
+    """
+    Test the `base64_decode` Jinja filter.
+    """
+    rendered = render_jinja_tmpl(
+        "{{ 'cmFuZG9t' | base64_decode }}",
+        dict(opts=minion_opts, saltenv="test", salt=local_salt),
+    )
+    assert rendered == "random"
+
+
+def test_json_query(minion_opts, local_salt):
+    """
+    Test the `json_query` Jinja filter.
+    """
+    rendered = render_jinja_tmpl(
+        "{{ [1, 2, 3] | json_query('[1]')}}",
+        dict(opts=minion_opts, saltenv="test", salt=local_salt),
+    )
+    assert rendered == "2"
+
+
+# def test_print():
```

Comment:
What's this one about?

Reply:
No idea. Was commented out in the original.  Deleting.